### PR TITLE
AYS-569 | Added permission control to forgot password service

### PR DESCRIPTION
**Pull request by sarpkan00**
### Checklist

Before submitting your pull request, ensure the following:

- [ ] **Title and Branch Naming Conventions:**
    - The pull request title follows the
      standard: [Pull Request Naming Conventions](https://github.com/afet-yonetim-sistemi/ays-be/blob/main/CONTRIBUTING.md#pull-request-naming-conventions).
    - The branch name follows one of the conventions outlined in
      the [Branch Naming Conventions](https://github.com/afet-yonetim-sistemi/ays-be/blob/main/CONTRIBUTING.md#branch-naming-conventions).

- [ ] **Local Testing:**
    - I have tested my changes locally on Postman, and they are working as expected.

- [ ] **Code Quality:**
    - The code is formatted according to the project's coding guidelines and style.
    - The code has been reviewed to ensure its quality.
    - The code does not contain any issues flagged by SonarLint.

- [ ] **Documentation:**
    - Necessary documentation has been added or existing documentation has been updated, specifically detailing changes made in Postman.

- [ ] **Testing:**
    - Relevant unit tests have been written and included.
    - Relevant integration tests have been written and included.

- [ ] **Reviewers and Assignees:**
    - Default reviewers have been assigned to this pull request.
    - Assignees have been added if necessary.

- [ ] **Labels and Associations:**
    - No specific actions are required in the Labels and Associations section for this pull request.

### CODE REVIEW

**Comment by agitrubard** [comment]
@sarpkan00 diğer arkadaşları da reviewer olarak ekleyebilir miyiz? Ek olarak checklist kontrol edilmemiş gibi duruyor. Kontrol ettikten sonra tikleri işaretleyebilir misin? Editleyerek manuel işaretlersen daha iyi olur, pull request update olunca her seferinde yeni bir pipeline başlatıyor

File: src/test/java/org/ays/auth/controller/AysAuthControllerTest.java [hunk 1]
```diff
@@ -249,7 +249,7 @@ void givenValidForgotPasswordRequest_whenSendPasswordCreateMail_thenReturnSucces
         // Then
         String endpoint = BASE_PATH.concat("/password/forgot");
         MockHttpServletRequestBuilder mockHttpServletRequestBuilder = AysMockMvcRequestBuilders
-                .post(endpoint, mockForgotPasswordRequest);
+                .post(endpoint, mockAdminToken.getAccessToken(), mockForgotPasswordRequest);
```
Comment (agitrubard) [inline src/test/java/org/ays/auth/controller/AysAuthControllerTest.java:252]:
```suggestion
                .post(endpoint, mockForgotPasswordRequest);
```

**Review by agitrubard — CHANGES_REQUESTED** [review]
Test tarafına daha sonra bakacağım, diğer kısımları kontrol ettim

File: src/main/java/org/ays/auth/model/request/AysPasswordForgotRequest.java [hunk 2]
```diff
@@ -15,4 +17,7 @@ public class AysPasswordForgotRequest {
     @Size(min = 2, max = 255)
     private String emailAddress;
 
+    @NotNull
+    private AysSourcePage sourcePage;
+
```
[the remark this answers is not in the document]
Comment (sarpkan00) [reply src/main/java/org/ays/auth/model/request/AysPasswordForgotRequest.java:22]: token servisi ile benzer yaptım. Kod bütünlüğü açısından. Evet söylediğinde bir seçenek ama kod bütünlüğü açısından token servisinide değiştirmek mantıklı olur. Benzer kontrol bir yerde farklı bir yerde farklı yapılmış gibi olucak

**Pull request closed without merging**